src/backend/metal/src/factory.rs

```diff
@@ -247,6 +247,10 @@ impl core::Factory<Resources> for Factory {
                     .object_at(0)
                     .set_pixel_format(MTLPixelFormat::BGRA8Unorm_sRGB);
 
+                // We need fake depth attachments in case explicit writes to the depth buffer are required
+                pso_descriptor.set_depth_attachment_pixel_format(MTLPixelFormat::Depth24Unorm_Stencil8);
```

Comment:
Good catch, apparently D24S8 is only supported on *some* OS X GPUs (and not at all on iOS). According to the docs all feature sets support Depth32_Stencil8 so that's a good choice.